tex from tetex segfaults

## Issue description

The `tex` program from the `tetex` package segfaults.

### Steps to reproduce

Run:

``` shell
$ nix-shell --pure -p tetex --run tex
```

type anything, then return:

```
This is TeX, Version 3.141592 (Web2C 7.5.4)                                                                                                                                                
**something                                                                                                                                                                          
/run/user/1000/nix-shell.s4CXAR/rc: line 1: 15467 Segmentation fault      tex   
```

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.63, NixOS, 17.09.2138.48bf1f43cd (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(thblt): `""`
 - channels(root): `"nixos-17.09.2138.48bf1f43cd"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

@orivej You're right, but building auctex depends on tetex somehow.  I hit the problem when trying to build `release_11_91`:

```
$ nix-shell --pure -p gnumake   texlive.combined.scheme-full --run make                                                          (cd latex ; make all)                                                                                                                                                                      
make[1]: Entering directory '/home/thblt/.emacs.d/lib/auctex/latex'                                                                                                                        
/nix/store/kgnahyg1jp2230n93cqzh2l7ykwgxnmw-tetex-3.0/bin/tex '\nonstopmode \input bootstrap.ins'
This is TeX, Version 3.141592 (Web2C 7.5.4)                                                                                                                                                
make[1]: *** [Makefile:74: preview-mk.ins] Segmentation fault                                                                                                                              
make[1]: Leaving directory '/home/thblt/.emacs.d/lib/auctex/latex'
make: *** [Makefile:214: texmf] Error 2
```

--- a/pkgs/tools/typesetting/tex/tetex/default.nix
+++ b/pkgs/tools/typesetting/tex/tetex/default.nix
@@ -26,7 +26,7 @@ stdenv.mkDerivation {
     find ./ -name "config.guess" -exec rm {} \; -exec ln -s ${automake}/share/automake-*/config.guess {} \;
   '' else null;
 
-  patches = [ ./environment.patch ./getline.patch ./clang.patch ];
+  patches = [ ./environment.patch ./getline.patch ./clang.patch ./extramembot.patch ];
 
   setupHook = ./setup-hook.sh;
 

--- a/pkgs/tools/typesetting/tex/tetex/extramembot.patch
+++ b/pkgs/tools/typesetting/tex/tetex/extramembot.patch
@@ -1,0 +1,12 @@
+diff -up texlive-2007/texk/web2c/tex.ch.extramembot texlive-2007/texk/web2c/tex.ch
+--- texlive-2007/texk/web2c/tex.ch.extramembot	2006-12-19 02:11:11.000000000 +0100
++++ texlive-2007/texk/web2c/tex.ch	2011-11-30 12:03:32.052795763 +0100
+@@ -365,7 +365,7 @@ for i:=@'177 to @'377 do xchr[i]:=i;
+ {Initialize enc\TeX\ data.}
+ for i:=0 to 255 do mubyte_read[i]:=null;
+ for i:=0 to 255 do mubyte_write[i]:=0;
+-for i:=0 to 128 do mubyte_cswrite[i]:=null;
++for i:=0 to 127 do mubyte_cswrite[i]:=null;
+ mubyte_keep := 0; mubyte_start := false; 
+ write_noexpanding := false; cs_converting := false;
+ special_printing := false; message_printing := false;

